lib/portage/sync/modules/git/git.py

```diff
@@ -207,19 +207,17 @@ def verify_head(self, revision='-1'):
 				'sync-git-verify-commit-signature', 'false').lower() not in ('true', 'yes')):
 			return True
 
-		if self.repo.sync_openpgp_key_path is not None:
-			if gemato is None:
-				writemsg_level("!!! Verifying against specified key requires gemato-11.0+ installed\n",
+		if self.repo.sync_openpgp_key_path is not None and gemato is None:
+			writemsg_level("!!! Verifying against specified key requires gemato-11.0+ installed\n",
 					level=logging.ERROR, noiselevel=-1)
-				return False
-			openpgp_env = gemato.openpgp.OpenPGPEnvironment()
-		else:
-			openpgp_env = None
+			return False
+
+		openpgp_env = self._get_openpgp_env(self.repo.sync_openpgp_key_path)
 
 		try:
 			out = EOutput()
 			env = None
-			if openpgp_env is not None:
+			if openpgp_env is not None and self.repo.sync_openpgp_key_path is not None:
```

Comment:
@mgorny I handled the case here where `openpgp_env` was `None` by checking if key is provided. By default the common function returns the system env if the key is not provided as you suggested. I don't see any other corner case where a `None` value would impact something. Do you?